scrapers_next/wv/committees.py

```diff
@@ -0,0 +1,151 @@
+from re import sub
+from spatula import HtmlPage, HtmlListPage, CSS, URL, Source, SkipItem
+from openstates.models import ScrapeCommittee
+
+
+class CommitteeMember(HtmlPage):
+    """
+    Scrape a committee member profile, if existing.
+
+    Position is scraped from the directory page and passed directly to this class.
+
+    If a member does not have a profile or their name cannot be gleaned using a
+    css selector, use fallback_name.
+
+    :param position: Rank of the member in the committee currently being scraped.
+    e.g. "Chair"
+    :param fallback_name: The name to use if the legislator's full name cannot be
+    determined from their profile. e.g. "Senator Burns"
+    """
+
+    position: str
+    fallback_name: str
+
+    def __init__(self, input_val, source: Source, position: str, fallback_name: str):
+        super(CommitteeMember, self).__init__(input_val, source=source)
+        self.position = position
+        self.fallback_name = fallback_name
+
+    def process_page(self):
+        # min_items set to zero because not all members have a profile
+        # page instead shows a list of media releases
+        name_match = CSS("#wrapleftcolr > h2:nth-child(1)", min_items=0).match(
+            self.root
+        )
+        name = name_match[0].text if len(name_match) == 1 else self.fallback_name
+        self.input.add_source(self.source.url, f"{name} url")
+        self.input.add_member(sub(r"\(.*\)", "", name), self.position)
+        return self.input
+
+    def get_source_from_input(self):
+        pass
+
+
+class CommitteeDetail(HtmlListPage):
+    """
+    Scrape a list of committee members.
+
+    :param selector: CSS matcher for each legislator url, according to chamber.
+    :param url_signifier: Part of each matched element's href that should exist
+    in a valid member profile url.
+    """
+
+    selector: CSS
+    url_signifier: str
+
+    def __init__(
+        self,
+        input_val: ScrapeCommittee,
+        source: Source,
+        selector: CSS,
+        url_signifier: str,
+    ):
+        super(CommitteeDetail, self).__init__(input_val, source=source)
+        self.selector = selector
+        self.url_signifier = url_signifier
+
+    def process_item(self, item):
+        if self.url_signifier in item.get("href"):
+            next_elem = item.getnext().text
+            position = next_elem if isinstance(next_elem, str) else "member"
+            return CommitteeMember(
+                self.input,
+                source=URL(item.get("href")),
+                position=position,
+                fallback_name=item.get("href").split("member=", 1)[1],
+            )
+        else:
+            raise SkipItem("not a committee member url")
+
+    def get_source_from_input(self):
```

Comment:
This is a required override for any class extending `HtmlPage` or `HtmlListPage`. The script will run without it, but I was getting warnings in my editor. Another way is to add `ABC` as a class extension.

Reply:
Interesting! I've not seen that warning before, and I just commented out those method from the code on my local (in PyCharm) and ran it and I'm not getting any warnings.

1. Which editor are you using?
2. Can you include a screenshot or copy past the warning you're getting? It would be helpful to know.

I appreciate it!

Reply:
Yeah it is interesting - I was able to run everything without these methods as well, so it's obviously not a requirement for running the application. The [declaration in Spatula's source](https://github.com/jamesturk/spatula/blob/9ff3678740eec19ae528aafc5f4f12d41c2f35b8/src/spatula/pages.py#L280) is identical to the other base methods, so this has to be a local config issue.

I'm using IntelliJ IDEA Ultimate with Python plugins. Interpreter is set to the local poetry version. 

For some reason GH isn't letting me upload a screenshot at the moment, but the message I'm getting is 
```
Class CommitteeMember must implement all abstract methods
```
The quick fix option is to `Add abc.ABC to all superclasses` - which results in `CommitteeMember(HtmlPage, ABC)`

Anyway, I'll remove these in an upcoming commit!

Reply:
Removal of methods: [c75e495](https://github.com/openstates/openstates-scrapers/pull/4398/commits/c75e495355fb42fb500885f7e746621da1f2226c) - though be sure to note the above about the `NotImplementedError` just in case that's a potential issue.

Reply:
Hey @chrisyamas I believe this is ready for another pass from you when you have a moment. Thanks!